### ISSUE TITLE
NiceHash

### ISSUE THREAD

**Issue by OptimusBlue** [dvandal/cryptonote-nodejs-pool#191]
Hi,

When trying to mine from with NiceHash it will connect but no hashing will take place, Difficulty will just continue to adjust lower.

What is required to make this NiceHash compatible or to resolve this issue?

Thanks

**Comment by mannie65** [comment]
Which coin is this with ? If its with ETN its not the pool software

**Comment by OptimusBlue** [comment]
It is with ETN, but the ETN pool on Nonopool is working.

**Comment by mannie65** [comment]
I run 7 pools with this software, all work wih nicehash apart from ETN. Nanopool have applied some additional patches i think as they work directly with the devs. I am under the opinion that post fork 1. Their algo changes have caused this or 2. Its got something to do with the handling of high diff zero blocks

**Comment by OptimusBlue** [comment]
Okay thanks for the info, I will relay it to the ETN Dev team.
Seems like fairhash.org is also able to take NiceHash

**Comment by mannie65** [comment]
Let me know how you go - and just as an FYI - even the electroneum pool code on their github doesn't work with nicehash. I thought is was this pool software so did and install of it and NO luck. Also ethos rigs running Claymore-XMR have exactly the same issue - they connect but wont submit hashes. This is very strange, when I approached the online Telegram admins, they just fobbed me off

**Comment by OptimusBlue** [comment]
Thanks, I have send this thread on to them so I hope they will look into it. What does not make sense is that I do know of ETN pools (only 2 at this point) that are NiceHash compatible, but they are not using this pool software. So where would the difference be?

**Comment by mannie65** [comment]
Well I know that using this pool software and the latest github release from ETN, both my ETHOS rigs and nicehash worked on my pool right up to block 307500, straight after that with No changes other than adding variant 1 to the pool and restarting - it all stopped working.

**Comment by mannie65** [comment]
I think I have found the issue - just need to test and validate it - will update tomorrow

**Comment by OptimusBlue** [comment]
Great news, looking forward to hear your findings. Thanks

**Comment by mannie65** [comment]
OK gents - I am still waiting to here from the ETN developers , but that might take a while so I will detail what I have found. Like other I applied the May 2018 V6 updates from ETN to my pool, based on this release. Everything was fine until block 307500 (V6 Fork) at which point My ETHOS rigs running Claymore-XMR 11.3 stop working. We also got complaints from Nicehash users that they could no longer submit NH to our pool. I thought it might have been an issue with this pool software post fork so I then installed a pool based on ETN own GitHub release, and had the exact same issues. This issue DOES NOT exist in any of our other Cryptonight pools that have forked. Anyway after a lot of testing and troubleshooting we identified that the problem goes away if we used ETN integrated wallet address, as opposed to a standard ETN address. This resolved the issue with both Nicehash and ETHOS/Claymore. Why I am not 100% sure - would love to hear from the ETN devs, but the problem ONLY happens with ETN v6 post fork.

Workaround - use ETN integrated addresses
